lua/tree-sitter-jinja2.lua

```diff
@@ -1,16 +1,37 @@
 local M = {}
 
+function M.register_parser()
+  local ok, parsers = pcall(require, "nvim-treesitter.parsers")
+  if not ok then
+    return
+  end
 
-local install_info = {
-  url = "https://github.com/geigerzaehler/tree-sitter-jinja2",
-  files = { "src/parser.c" },
-  branch = "main",
-}
+  -- Register the plugin path as a tree-sitter parser to nvim-treesitter.
+  -- This is not required, but it allows nvim-treesitter to manage it e.g. :TS{Install,Update} jinja2
+  local file = debug.getinfo(1).source:match("@(.*/)")
+  local plugin_dir = vim.fn.fnamemodify(file, ":p:h:h:h")
+
+  --- @type ParserInfo[]
+  local parser_config
+
+  if type(parsers.get_parser_configs) == "function" then
+    parser_config = parsers.get_parser_configs()
+  else
+    -- "nvim-treesitter.parsers" exports the parser table itself starting nvim-treesitter v1.x
+    parser_config = parsers
+  end
+
+  if type(parser_config.ghostty) == "table" then
+    -- A jinja2 parser is already registered. Avoid overwriting it.
+    return
+  end
```

Comment:
I don’t think ghostty has anything to do with this plugin. Even there is a jinja2 parser already installed I would prefer to override it.

```suggestion
```

Reply:
Thanks - I made a similar change to `tree-sitter-ghostty` which leaked over.. 🫠